### DISCUSSION TITLE
v.3.2 - New features and custom configurations

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#467] [Announcements]
### As always, we would like to thank everyone who donated to this project and remind you that every donation helps to support this project - so if you haven't made a donation yet, now would be a good time ;)

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

&nbsp;

### However, we have managed to add some features that you have been asking for.

&nbsp;
&nbsp;

### Since in this update also the Blueprint was changed and some things were deleted because new fields were added - please check your Blueprint if all fields are still filled (it can happen that some fields are empty after the update).

&nbsp;

> ### ⚠️ If you have issues installing the tft-file, DO NOT create a new isse, read here 
> [Common issues](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki) / https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/166 / https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/177 / https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/197
> ### Or use the Issues search: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues?q=is%3Aissue+tft+is%3Aclosed
> ### This is not a bug in this project, this is how the implemetion of nextion in esphome just works

&nbsp;

## Files that need to be reloaded:

1. nspanel_esphome.yaml
2. nspanel_blueprint.yaml
3. nspanel.tft

 _Update procedure can be read here:_

EN: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know

DE: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(DE)-HowTo---Alle-wichtigen-Dinge-die-man-wissen-sollte

&nbsp;

## Community Updates
- Add Hebrew language - thanks @leranp
- Convert "Update TFT display" into a button - thanks @edwardtfn
- Fix default language input - thanks @edwardtfn
- thanks for the great support and support in HA Community Forum - thanks @edwardtfn
- reformat code - remove obsolete spaces - thanks @bakito and @chpego
- Fix so homepage works when one of home values 01-03 is unavailable - thanks @Floppe
- Fix warning on settings_entity_value - thanks @edwardtfn

&nbsp;

## Overview of all changes

1. **_"Multi climate"_** - The climate entity can now be selected for all 32 buttons.

2. **_"Add button icon and icon color"_** - You can now set your own icons and colors for the buttons.

3. **_"Add chips on the homepage"_** - Seven custom chips can now be selected.

4. **_"3 new fields on the homepage"_** - Three custom sensors can now be selected.

5. **_"Add custom action for HW Buttons"_** - Custom actions can now be defined for the hardware buttons.

6. **_"Wifi Icon"_** - A Wi-Fi icon is displayed when the NSPanel is connected to Wi-Fi.

7. **_"Custom homepage"_** - On the home page, all icons, icon colors, and text colors can be customized.

8. **_"Rework Slider"_** - Issues with the light slider and cover slider have been fixed.

9. **_"Rework US version design"_** - The US version has received design adjustments, including the "wave design."

10. **_"Settings page Icon"_** - The settings icon has been removed and replaced by a "hidden" button on the clock.

11. **_"New Blueprint design"_** - The Blueprint design has been updated.

12. **_"Bug fixes and improvements in the blueprint, ESPHome, and TFT code."_** - Bug fixes and improvements have been made to the Blueprint, ESPHome, and TFT code.

You can find all the icons that can be used at this link:
https://htmlpreview.github.io/?https://github.com/jobr99/Generate-HASP-Fonts/blob/master/cheatsheet.html

&nbsp;

### 1. Multi climate
Climate Entity can now be selected from the 32 buttons. The current temperature is displayed on the button.
With a short or long press you jump to the climate page and can control and adjust the climate entity

![Github_EU-1-4](https://user-images.githubusercontent.com/41958506/226168043-d360b11f-6c8e-4340-aba5-71affe175b6d.png)

&nbsp;

### 2. Add button icon and icon color
You can now store your own icons in the blueprint for the buttons. In addition, the color of the icons can be selected.
If no custom icon is selected, a default icon and icon color will be set to match the selected entity

However, there is a small disadvantage. Currently, the page takes a few milliseconds longer to load than before.
This is because every icon, font button, etc. is reloaded whenever you swipe to this page.
The new layout of the page was necessary due to the adjustments of the custom icons and icon colors, and because we had to remove all "global" variables from the Nextion Editor to reduce the memory usage, otherwise no further functions could be built in.

![Github_EU-1](https://user-images.githubusercontent.com/41958506/226168199-50dac0a7-02d3-40b9-9b01-4c0781ed9e02.png)

&nbsp;

### 3. Add chips on the homepage
7 chips can now be configured in the blueprint. These chips are displayed at the top of the home page.
The chips only work with entities that have an on or off status. Therefore the following domains are allowed:
- binary_sensor
- sensor
- light
- switch

A separate icon and icon color can be stored in the blueprint for the chips.

![Github_EU-1-7](https://user-images.githubusercontent.com/41958506/226168209-42358090-d0fb-4e88-8d17-63893e51ac95.png)

&nbsp;

### 4. 3 new fields on the homepage
We deleted the "humidity" and "hotwater" fields on the home page.
For this we have added 3 free sensor fields. You can now store your own sensors that should be displayed.

Here too, the icons, icon color, and label color can be changed.

![Github_EU-1-6](https://user-images.githubusercontent.com/41958506/226168220-645c2f7c-98ef-4e76-b8bb-c4cd672d5bdb.png)

&nbsp;

### 5. Add custom action for HW Buttons
A custom action can now be stored in the blueprint for the long press of the hardware buttons
You can now choose whether you want to store a default action such as light settings or a custom action

![Bildschirmfoto 2023-03-19 um 12 55 49](https://user-images.githubusercontent.com/41958506/226173548-f62391df-9179-4b9d-8e99-a527f67c6ff8.png)

&nbsp;

### 6. Wifi Icon
A Wifi icon has now been added next to the time display. This WiFi icon indicates whether your NSPanel is connected to the WLAN. If the connection to WiFi is lost, no icon will be displayed.
The connection is checked every 60 seconds.

![Github_EU-1-5](https://user-images.githubusercontent.com/41958506/226168236-c64ea56e-6569-4a1f-9f8a-be2df86f1f6e.png)

&nbsp;

### 7. Custom homepage
A lot has changed on the home page so that you can customize your NSPanel to your liking.
It is now possible to replace every default icon with your own icon via the blueprint.
In addition, the color for each icon or text can now be changed.
All changes are available in the blueprint.

![Github_EU-1-3](https://user-images.githubusercontent.com/41958506/226168254-e3d7be56-f2ca-4830-b1b2-96025ecb2e4e.png)

&nbsp;

### 8. Rework Slider
The sliders for brightness, color temperature, and cover have caused us problems time and time again. This was due to the synchronization with HA.
We have now revised the logic for the sliders. This also reduces the load on HA and the NSPanel.

When a page with the slider is loaded, the information is first retrieved from HA.
When the slider is moved, the new state is only transmitted to HA once the slider is released. This has the advantage that now only one value is transmitted to HA.

&nbsp;

### 9. Rework US version design
Due to the many changes, we had to revise the design again and have now incorporated the "Wave" on some pages as on the EU version.

![Github_US-1-14](https://user-images.githubusercontent.com/41958506/226174331-ead2a1d6-05b0-4295-b612-3499cf5b344b.png)

&nbsp;

### 10. Settings page Icon
The Settings Icon was removed from the home page because there was no more space. On the other hand, we achieved a clean design for the homepage.

Another reason for this change was that we wanted to "hide" the settings so that they are not easily visible or accidentally clicked on by children or guests.

But don't worry, the function is not gone. The settings page can now be accessed via a "Long Press" on the clock.

![Github_EU-1-9](https://user-images.githubusercontent.com/41958506/226168325-6b0ec0c5-ff97-476d-8909-eff90ba768a5.png)

&nbsp;

### 11. New Blueprint design
This was an important point for us that was long overdue.
As you know, from the beginning of the project, it was important for us that it can be used and configured as easily as possible for everyone.

Due to the many new features, the blueprint became very long and unclear.

Therefore, we have adjusted the descriptions of the fields so that it is clearer what this field is for and also for which page. In addition, we have added some "headings". These do not have any functions but serve only for better overview. Especially when you have to scroll quickly through the blueprint because you are looking for something specific, it was always difficult to see where you are. With the new headings, it is now clear where you are in the blueprint.

![Bildschirmfoto 2023-03-19 um 11 09 00](https://user-images.githubusercontent.com/41958506/226168576-86a5c9cb-7c8c-4582-a74b-f55e13a548d5.png)

&nbsp;

### 12. Bug fixes and improvements in the blueprint, ESPHome, and TFT code.
As with every release, we have made some adjustments and fixes to the code to provide a better experience with our NSPanel version.

Adjustments to the blueprint and ESPHome have caused issues due to the latest updates of HA and ESPHome.

&nbsp;
&nbsp;

## Next topics we are currently working on
See here: https://github.com/Blackymas/NSPanel_HA_Blueprint/labels/roadmap

## What's Changed
* Update README.md by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/194
* Update Readme by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/195
* change tft baud to bauds by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/198
* change tft baud to bauds by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/199
* Simplify and remove useless code in ESP32 by @X-Ryl669 in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/207
* Update nspanel_esphome.yaml in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/231
* QR-code, local fallback by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/233
* upd: set boot-infos by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/239
* upd: trigger on automation-reload by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/240
* small changes by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/250
* upd: entities by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/261
* upd: instant update of home-buttons by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/264
* Blackymas test sponsor by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/270
* upd: page-timer esphome by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/272
* fix: exclude screensaver by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/278
* upd: remove global variable touchevent by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/279
* upd: HW-buttons hold by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/282
* fix: new tft ids by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/283
* fix: unavailable entity by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/284
* upd: hw-button hold by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/289
* upd: change entity icon description by @deejaybeam in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/295
* rdy for Release by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/297
* v.3.1.0 - New features and US Version by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/298

## New Contributors
* @X-Ryl669 made their first contribution in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/207

**Full Changelog**: https://github.com/Blackymas/NSPanel_HA_Blueprint/compare/v.3.0.0...v.3.2

<hr /><em>This discussion was created from the release <a href='https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v.3.2'>v.3.2 - New features and custom configurations</a>.</em>

**Comment by lloydbayley** [comment]
It's great, I love it! Thank you for your efforts!

**Reply by Blackymas** [reply]
Thanks 🎉

**Comment by Rep-Hueman** [comment]
the blueprint is stil version 3.1.0 ??

**Reply by Blackymas** [reply]
Nope 3.2

**Reply by Rep-Hueman** [reply]
When I download the zip file at the bottom of 3.2 --  (NSPanel_HA_Blueprint-v.3.2.zip) it is the old code from 3.1.0. 
I have now used the files from main (NSPanel_HA_Blueprint-main.zip) this is version 3.2.

**Comment by sneekerzzz** [comment]
Nice Release!! Where to find the color codes? I don't see that in the docs?

**Reply by edwardtfn** [reply]
You can Google for RGB565, with is the color palette used by the panel.

Here is a RGB565 color picker: http://www.barth-dev.de/online/rgb565-color-picker/

**Reply by Blackymas** [reply]
> Nice Release!! Where to find the color codes? I don't see that in the docs?

https://nodtem66.github.io/nextion-hmi-color-convert/index.html

**Reply by bakito** [reply]
I created a PR to add links to color converter and icons to the blueprint https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/476

**Comment by edwardtfn** [comment]
I've tested the chips, all working fine.Be aware that chips are visible in the panel when the entity's status is on, otherwise the chip is not visible.On 19 Mar 2023, at 17:57, geni0602 ***@***.***> wrote:﻿
Hi,
nice release!
Did somebody tested the new chips? I configured them on 2 NSpanels, but they are not visible after automation is saved. I´ve tested with a light and binery sensor.

**Reply by geni0602** [reply]
I can confirm everything is working fine. I notice it, after my comment. They are visible, only when the entity is on

**Reply by Blackymas** [reply]
Perfect 👍

**Comment by Rep-Hueman** [comment]
Hi Blackymas,
What a great job. all works great also with 3 panels here! version 3.2 now!

I just made you a nice donation for your work. €20

**Reply by Blackymas** [reply]
Wow that makes me happy and thank you very much for the donation 😍

**Comment by robgough1970** [comment]
fantastic work guys! installed the beta a few days ago and the new features are brilliant. so easy to customise. just posted the link on the HA FB to make others aware of this great project. 👍

**Reply by Blackymas** [reply]
Can you share the link?
Big thanks!!!

**Reply by robgough1970** [reply]
no problem 
https://www.facebook.com/groups/HomeAssistant/permalink/3465568947047784/

**Comment by sqalg** [comment]
Hello

Many thanks for your great work.
A very mature approach .

**Reply by Blackymas** [reply]
thanks ;)
